src/omero_zarr/raw_pixels.py

```diff
@@ -245,7 +256,14 @@ def plate_to_zarr(plate: omero.gateway._PlateWrapper, args: argparse.Namespace)
     total = n_rows * n_cols * (n_fields[1] - n_fields[0] + 1)
 
     target_dir = args.output
-    name = os.path.join(target_dir, "%s.zarr" % plate.id)
+    name_by = args.name_by
+
+    if name_by == "name":
```

Comment:
Two quick notes:
 - this code is duplicated and could be made a method for possible sub-classing
 - I can live with this split in naming schemes, but it could be a surprise